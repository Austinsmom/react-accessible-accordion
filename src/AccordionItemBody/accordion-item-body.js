import React, { PropTypes } from 'react';
import classNames from 'classnames';

const defaultProps = {
    id: '',
    expanded: false,
    className: 'accordion__body',
    prefixCss: 'accordion__body',
    role: '',
};

const propTypes = {
    id: PropTypes.string,
    expanded: PropTypes.bool,
    children: PropTypes.oneOfType([
        PropTypes.element,
        PropTypes.arrayOf(PropTypes.element),
    ]).isRequired,
    className: PropTypes.string,
    prefixCss: PropTypes.string,
    role: PropTypes.string,
};

const AccordionItemBody = ({ id, expanded, children, className, prefixCss, role }) => {
    const bodyClass = classNames(
        className,
        {
            [`${prefixCss}--hidden`]: !expanded,
        },
    );
    const ariaHidden = !expanded;
    return (
        <div
            id={id}
            className={bodyClass}
            aria-hidden={ariaHidden}
            role={role}
        >
            {children}
        </div>
    );
};

AccordionItemBody.propTypes = propTypes;
AccordionItemBody.defaultProps = defaultProps;

export default AccordionItemBody;
