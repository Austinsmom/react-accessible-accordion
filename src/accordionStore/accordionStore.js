// @flow

import { observable, action } from 'mobx';

export type Item = {
    itemkey: string | number,
    itemUuid: number,
    expanded: boolean,
    disabled: boolean,
};
export type Store = {
    items: Array<Item>,
    accordion: boolean,
    onChange: Function,
    addItem: Item => void,
    removeItem: ($PropertyType<Item, 'itemkey'>) => void,
    setExpanded: ($PropertyType<Item, 'itemkey'>, boolean) => void,
};

// eslint-disable-next-line import/prefer-default-export
export const createAccordionStore = ({
    accordion,
    onChange,
}: {
    accordion: boolean,
    onChange: Function,
}): Store => {
    const store: Store = observable({
        items: [],
        accordion,
        onChange,
        addItem: action.bound(function addItem(item) {
            this.items = [...this.items, item];
        }),
        removeItem: action.bound(function removeItem(key) {
            this.items = this.items.filter(item => item.itemkey !== key);
        }),
        setExpanded: action.bound(function setExpanded(
            key: string | number,
            expanded: boolean,
        ) {
            this.items = this.items.map(item => {
                if (item.itemkey === key) {
                    return {
                        ...item,
                        expanded,
                    };
                }
                return item;
            });
        }),
    });
    return store;
};
