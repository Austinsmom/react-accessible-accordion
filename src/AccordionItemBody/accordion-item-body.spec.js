// @flow

import React from 'react';
import { Provider } from 'mobx-react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { createAccordionStore } from '../accordionStore/accordionStore';
import AccordionItemBody from './accordion-item-body';

describe('AccordionItemBody', () => {
    let accordionStore;
    const uuid = 'asdf-1234';

    beforeEach(() => {
        accordionStore = createAccordionStore({
            accordion: true,
            onChange: jest.fn(),
        });
        accordionStore.addItem({
            uuid,
            expanded: false,
            disabled: false,
        });
    });

    it('renders correctly with min params', () => {
        const tree = renderer
            .create(
                <Provider accordionStore={accordionStore} uuid={uuid}>
                    <AccordionItemBody>
                        <div>Fake body</div>
                    </AccordionItemBody>
                </Provider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with different className', () => {
        const className = 'className';
        const wrapper = mount(
            <Provider accordionStore={accordionStore} uuid={uuid}>
                <AccordionItemBody className={className} />
            </Provider>,
        );
        expect(wrapper.find('div').hasClass(className)).toEqual(true);
    });

    it('renders correctly with different hideBodyClassName', () => {
        const hideBodyClassName = 'hideBodyClassName';
        const wrapper = mount(
            <Provider accordionStore={accordionStore} uuid={uuid}>
                <AccordionItemBody hideBodyClassName={hideBodyClassName} />
            </Provider>,
        );
        expect(wrapper.find('div').hasClass(hideBodyClassName)).toEqual(true);
    });

    it('renders null if an associated AccordionItem is not registered in accordionStore', () => {
        const className = 'className';
        const wrapper = mount(
            <Provider accordionStore={accordionStore} uuid="foo">
                <AccordionItemBody className={className}>
                    <div>Fake body</div>
                </AccordionItemBody>
            </Provider>,
        );

        expect(
            wrapper.findWhere(item => item.className === className).length,
        ).toEqual(0);
    });

    it('respects arbitrary user-defined props', () => {
        const wrapper = mount(
            <Provider accordionStore={accordionStore} uuid={uuid}>
                <AccordionItemBody lang="en" />
            </Provider>,
        );
        expect(wrapper.find('div').instance().lang).toEqual('en');
    });
});
