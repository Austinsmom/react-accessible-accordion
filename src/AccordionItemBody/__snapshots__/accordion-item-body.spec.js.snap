// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionItemBody renders correctly with different className 1`] = `
<div
  aria-hidden={true}
  className="testCSSClass accordion__body--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with expanded 1`] = `
<div
  aria-hidden={false}
  className="accordion__body"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with min params 1`] = `
<div
  aria-hidden={true}
  className="accordion__body accordion__body--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;

exports[`AccordionItemBody renders correctly with prefixClass 1`] = `
<div
  aria-hidden={true}
  className="accordion__body testCSSClass--hidden"
  id=""
  role=""
>
  <div>
    Fake body
  </div>
</div>
`;
